### ISSUE TITLE
[Feature] Add option in multipart form data to upload file or text

### ISSUE THREAD

**Issue by nidhi-nair** [appsmithorg/appsmith#5547]
## Description

To allow using file meta data, the server supports an object representation for file types. The API pane needs to allow users to choose the file type to be able to use this object. The property for each key-value pair would then look like:
```
{
  "key": "keyName",
  "value": "valueField",
  "type": "FILE" (or "TEXT")
}
```

**Comment by nidhi-nair** [comment]
The designs for this are available on: https://www.figma.com/file/ko05rfZRuc8HzqBZRpg1fY/API-Pane-Updates?node-id=119%3A2
Ref: #4907

**Comment by Nikhil-Nandagopal** [comment]
@nidhi-nair @hetunandu can we not read the content type from the file automatically? I don't think we should ask the user to set this

**Comment by hetunandu** [comment]
> @nidhi-nair @hetunandu can we not read the content type from the file automatically? I don't think we should ask the user to set this

@Nikhil-Nandagopal 
We will try to prefill this the best we can, but the field will be present for the user to update it

**Comment by Nikhil-Nandagopal** [comment]
@hetunandu I think the file is expected to have this data. I think a field is error prone and possibly a bad idea. Postman does not have a field so I don't see why we should. It just confuses the UI

**Comment by nidhi-nair** [comment]
@Nikhil-Nandagopal Adding more context:
The reason Postman works without exposing the field is that they only accept inputs for multipart data from the file explorer. Since the file metadata is available to them from this route, they are guaranteed to calculate the correct mime type. 

In our case, one source is the file picker, for which the above format is to be included via #5503. For other sources, the ability to include file data should be supported by the API pane is why the design includes the field.

**Comment by Nikhil-Nandagopal** [comment]
@nidhi-nair In that case should this be another field or should we include a complex file object field that has all the info. Is it possible we might need additional information like total file size as well? Maybe this field should expect a file object always
